src/main/java/duke/io/DukeParser.java

```diff
@@ -0,0 +1,109 @@
+package duke.io;
+
+import duke.type.DukeCommand;
+import duke.exception.DukeFileSystemException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * A class that handles anything related to the parsing of information for Duke.
+ */
+public class DukeParser {
+    /**
+     * Constructs a new instance of the DukeParser. No arguments are expected.
+     */
+    public DukeParser() {
+
+    }
+
+    /**
+     * Provides a task based on data read from line.
+     *
+     * @param line line that data is read from.
+     * @return a task matching the information from the line of data.
+     * @throws DukeFileSystemException if the line provided does not conform to the Duke storing format.
+     */
+    public Task parseTaskFromLine(String line) throws DukeFileSystemException {
+        String[] lineArgs = line.split(" \\| ");
+        String taskType = lineArgs[0];
+        boolean isDone = lineArgs[1].equals("1");
+        String taskDescription = lineArgs[2];
+        String date = lineArgs.length > 3 ? lineArgs[3] : "";
+        Task task;
+        switch (taskType) {
+        case "T":
+            task = new Todo(taskDescription);
+            break;
+        case "D":
+            task = new Deadline(taskDescription, date);
+            break;
+        case "E":
+            task = new Event(taskDescription, date);
+            break;
+        default:
+            throw new DukeFileSystemException("Invalid data read from file while loading tasks. A new list will "
+                    + "be used for this session.");
+        }
+        if (isDone) {
+            task.markAsDone();
+        }
+        return task;
+    }
+
+    /**
+     * Gets the type of Duke command of a given input string.
+     * The Duke command of an input string is given by the first argument provided in the string.
+     *
+     * @param inputString the string literal of the input.
+     * @return a corresponding Duke command type based on the input string.
+     */
+    public DukeCommand getCommandType(String inputString) {
+        DukeCommand inputCommand;
```

Comment:
Even for a long method like this, you still managed to avoid the arrowhead style. Made it way easier to read and understand. Good job!